velox/dwio/dwrf/common/CacheInputStream.cpp

```diff
@@ -157,6 +158,65 @@ void CacheInputStream::loadSync(dwio::common::Region region) {
     if (entry->isExclusive()) {
       entry->setGroupId(groupId_);
       entry->setTrackingId(trackingId_);
+      auto ssdCache = cache_->ssdCache();
+
+      if (ssdCache) {
+        auto& file = ssdCache->file(fileNum_);
+        auto ssdPin =
+            file.find(cache::RawFileCacheKey{fileNum_, region.offset});
+        if (!ssdPin.empty() && ssdPin.run().size() < entry->size()) {
+          LOG(INFO) << fmt::format(
+              "IOERR: Ssd entry for {}:{} {} b shorter than requested {}b",
+              entry->key().fileNum.id(),
+              entry->key().offset,
+              ssdPin.run().size(),
+              entry->size());
+          ssdPin.clear();
+        }
+        if (!ssdPin.empty()) {
```

Comment:
This method is very long. It would be nice to refactor the logic inside this "if" statement into a helper method.